communication/vehicleconnections/mavsdkstation.cpp

```diff
@@ -12,34 +13,41 @@ MavsdkStation::MavsdkStation(QObject *parent) : QObject(parent)
     mHeartbeatTimer.start(1000);
 
     mMavsdk.subscribe_on_new_system([this](){
+        static bool temporaryFixToStopOuterCallbackFromBeingCalledAgainBeforeTheInnerCallbackHasFinished = true;
         for (const auto &system : mMavsdk.systems()) {
-            if (!mVehicleConnectionMap.contains(system->get_system_id())) {
+            if (!mVehicleConnectionMap.contains(system->get_system_id()) && temporaryFixToStopOuterCallbackFromBeingCalledAgainBeforeTheInnerCallbackHasFinished) {
                 if (system->has_autopilot()) {
+                    temporaryFixToStopOuterCallbackFromBeingCalledAgainBeforeTheInnerCallbackHasFinished = false;
                     qDebug() << "MavsdkStation: detected system" << system->get_system_id() << "waiting for another heartbeat for initializing MavsdkVehicleConnection...";
 
                     // Wait for heartbeat using passthrough to instantiate vehicleConnection (mainly needed to get MAV_TYPE)
                     auto mavlinkPassthrough = new mavsdk::MavlinkPassthrough(system);
-                    mavlinkPassthrough->subscribe_message(MAVLINK_MSG_ID_HEARTBEAT, [this, system, mavlinkPassthrough](const mavlink_message_t &message){
-                        // unsubscribe from further heartbeats by deleting passthrough
-                        delete mavlinkPassthrough;
-
+                    mavlinkPassthrough->subscribe_message(MAVLINK_MSG_ID_HEARTBEAT, [this, system, mavlinkPassthrough](const mavlink_message_t &message) mutable {
```

Comment:
Is mutable used to get rid of a const? For the system maybe? In that case I would rather remove the const from the for loop (l. 17) and not use mutable here.

Reply:
It's used for mavlinkPassthrough = nullptr;